## 2024년 종별코드명 활용하여 상급, 종합 병원 등의 분포 확인

In [11]:
import seaborn as sns
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.express as px


In [12]:
DATA_PATH = '/Users/haeunpark/mid_project_data'

In [13]:
# 2024 데이터 출력
pd.set_option('display.max_columns', None)


df_2024 = pd.read_csv(f'{DATA_PATH}/2024의원통합데이터.csv', index_col=0)
df_2024

암호화요양기호    요양기관명 종별코드명  \
0       JDQ4MTYyMiM4MSMkMiMkMCMkMDAkNDgxMTkxIzMxIyQxIy...  뽀빠이연합의원    의원   
1       JDQ4MTYyMiM4MSMkMiMkMCMkMDAkNDgxMTkxIzMxIyQxIy...  뽀빠이연합의원    의원   
2       JDQ4MTYyMiM4MSMkMiMkMCMkMDAkNDgxMTkxIzMxIyQxIy...  뽀빠이연합의원    의원   
3       JDQ4MTYyMiM4MSMkMiMkMCMkMDAkNDgxMTkxIzMxIyQxIy...  뽀빠이연합의원    의원   
4       JDQ4MTYyMiM4MSMkMiMkMCMkMDAkNDgxMTkxIzMxIyQxIy...  뽀빠이연합의원    의원   
...                                                   ...      ...   ...   
407395  JDQ4MTYyMiM4MSMkMSMkMCMkMTMkNDgxOTYxIzUxIyQxIy...    온유한의원   한의원   
407396  JDQ4MTYyMiM4MSMkMSMkMCMkMTMkNDgxOTYxIzUxIyQxIy...    온유한의원   한의원   
407397  JDQ4MTYyMiM4MSMkMSMkMCMkMTMkNDgxOTYxIzUxIyQxIy...    온유한의원   한의원   
407398  JDQ4MTYyMiM4MSMkMSMkMCMkMTMkNDgxOTYxIzUxIyQxIy...    온유한의원   한의원   
407399  JDQ4MTYyMiM4MSMkMSMkMCMkMTMkNDgxOTYxIzUxIyQxIy...    온유한의원   한의원   

       설립구분코드명 시도코드명 시군구코드명      개설일자  성인중환자병상수  일반입원실병상수  소아중환자병상수  \
0           개인    대구   대구중구  20230424         0         0         0   
1           개인    대구   대구중구  20230424         0         0         0   
2           개인    대구   대구중구  20230424         0         0         0   
3           개인    대구   대구중구  20230424         0         0         0   
4           개인    대구   대구중구  20230424         0         0         0   
...        ...   ...    ...       ...       ...       ...       ...   
407395      개인    대구   대구북구  20230713         0         0         0   
407396      개인    대구   대구북구  20230713         0         0         0   
407397      개인    대구   대구북구  20230713         0         0         0   
407398      개인    대구   대구북구  20230713         0         0         0   
407399      개인    대구   대구북구  20230713         0         0         0   

        신생아중환자병상수  분만실병상수  수술실병상수  응급실병상수  물리치료실병상수  정신과폐쇄상급병상수  정신과폐쇄일반병상수  \
0               0       0       0       0        10           0           0   
1               0       0       0       0        10           0           0   
2               0       0       0       0        10           0           0   
3               0       0       0       0        10           0           0   
4               0       0       0       0        10           0           0   
...           ...     ...     ...     ...       ...         ...         ...   
407395          0       0       0       0         0           0           0   
407396          0       0       0       0         0           0           0   
407397          0       0       0       0         0           0           0   
407398          0       0       0       0         0           0           0   
407399          0       0       0       0         0           0           0   

        정신과개방상급병상수  정신과폐쇄개방병상수  격리병실병상수  무균치료실병상수  총의사수  의과일반의 인원수  의과인턴 인원수  \
0                0           0        0         0     1          0         0   
1                0           0        0         0     1          0         0   
2                0           0        0         0     1          0         0   
3                0           0        0         0     1          0         0   
4                0           0        0         0     1          0         0   
...            ...         ...      ...       ...   ...        ...       ...   
407395           0           0        0         0     1          0         0   
407396           0           0        0         0     1          0         0   
407397           0           0        0         0     1          0         0   
407398           0           0        0         0     1          0         0   
407399           0           0        0         0     1          0         0   

        의과레지던트 인원수  의과전문의 인원수  치과일반의 인원수  치과인턴 인원수  치과레지던트 인원수  치과전문의 인원수  \
0                0          1          0         0           0          0   
1                0          1          0         0           0          0   
2                0          1          0         0           0          0   
3                0          1          0         0           0    

In [14]:
# 종별 코드 무엇이 있는지 확인
unique_hospital_types = df_2024['종별코드명'].unique()


print("종별코드명에 포함된 값들:")
for hospital_type in unique_hospital_types:
    print(f"- {hospital_type}")

종별코드명에 포함된 값들:
- 의원
- 치과의원
- 한의원
- 한방병원
- 정신병원
- 병원
- 보건지소
- 요양병원
- 종합병원
- 치과병원
- 보건진료소
- 상급종합
- 보건소
- 보건의료원
- 조산원


# 시도별 병원 유형 분석 차트

In [15]:
# 병원 유형 분류 사전
hospital_groups = {
    '상급종합': '3차 의료',
    '종합병원': '2차 의료',
    '병원': '2차 의료',
    '치과병원': '2차 의료',
    '한방병원': '2차 의료',
    '요양병원': '2차 의료',
    '정신병원': '2차 의료',
    '보건의료원': '2차 의료',
    '의원': '1차 의료',
    '치과의원': '1차 의료',
    '한의원': '1차 의료',
    '보건소': '1차 의료',
    '보건지소': '1차 의료',
    '보건진료소': '1차 의료',
    '조산원': '특수의료'
}

# 중복 제거: 암호화요양기호를 기준으로 중복된 병원 제거
df_2024_unique = df_2024.drop_duplicates(subset=['암호화요양기호'])

# 병원 유형에 따른 1차, 2차, 3차 의료 수준 분류 추가
df_2024_unique['의료수준'] = df_2024_unique['종별코드명'].map(hospital_groups)

# 시도별 병원 유형 분포 집계 (2024년도 데이터)
hospital_2024_by_region_level = df_2024_unique.groupby(['시도코드명', '의료수준']).size().unstack(fill_value=0)


df_2024_unique

/var/folders/zj/p3cl2py51ldfrgsbtfp578x40000gn/T/ipykernel_41521/3109706404.py:24: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_2024_unique['의료수준'] = df_2024_unique['종별코드명'].map(hospital_groups)


암호화요양기호       요양기관명 종별코드명  \
0       JDQ4MTYyMiM4MSMkMiMkMCMkMDAkNDgxMTkxIzMxIyQxIy...     뽀빠이연합의원    의원   
7       JDQ4MTYyMiM1MSMkMSMkMCMkNzIkNTgxMzUxIzUxIyQxIy...     청담e치과의원  치과의원   
17      JDU4MTI3MSM1MSMkMiMkMCMkMDAkNTgxOTYxIzIxIyQxIy...   연세웰드림내과의원    의원   
25      JDQ4MTYyMiM1MSMkMSMkMCMkNzIkNTgxOTYxIzIxIyQxIy...  연세바로세움치과의원  치과의원   
36      JDQ4MTYyMiM2MSMkMSMkMiMkNzIkMzgxMTkxIzIxIyQxIy...      다은치과의원  치과의원   
...                                                   ...         ...   ...   
407363  JDQ4MTYyMiM2MSMkMSMkMiMkMTMkNDgxOTYxIzIxIyQxIy...      감초당한의원   한의원   
407371  JDQ4MTYyMiM4MSMkMiMkMCMkMDAkNDgxMTkxIzMxIyQxIy...   더웰이비인후과의원    의원   
407376  JDQ4MTYyMiM4MSMkMSMkMCMkNzIkMzgxMTkxIzQxIyQxIy...     행복한치과의원  치과의원   
407384  JDQ4MTYyMiM4MSMkMSMkMCMkMTMkNDgxOTYxIzUxIyQxIy...       손길한의원   한의원   
407392  JDQ4MTYyMiM4MSMkMSMkMCMkMTMkNDgxOTYxIzUxIyQxIy...       온유한의원   한의원   

       설립구분코드명 시도코드명 시군구코드명      개설일자  성인중환자병상수  일반입원실병상수  소아중환자병상수  \
0           개인    대구   대구중구  20230424         0         0         0   
7           개인    경기    구리시  20190401         0         0         0   
17          개인    경기  성남수정구  20240112         0         0         0   
25          개인    경기  고양덕양구  20240517         0         0         0   
36          개인    대전   대전중구  20240117         0         0         0   
...        ...   ...    ...       ...       ...       ...       ...   
407363      개인    충남    보령시  20230707         0         0         0   
407371      개인    경북    구미시  20100420         0         0         0   
407376      개인    경북   포항북구  20130417         0         0         0   
407384      개인    대구  대구수성구  20230608         0         0         0   
407392      개인    대구   대구북구  20230713         0         0         0   

        신생아중환자병상수  분만실병상수  수술실병상수  응급실병상수  물리치료실병상수  정신과폐쇄상급병상수  정신과폐쇄일반병상수  \
0               0       0       0       0        10           0           0   
7               0       0       0       0         0           0           0   
17              0       0       0       0         0           0           0   
25              0       0       0       0         0           0           0   
36              0       0       0       0         0           0           0   
...           ...     ...     ...     ...       ...         ...         ...   
407363          0       0       0       0         0           0           0   
407371          0       0       0       0         0           0           0   
407376          0       0       0       0         0           0           0   
407384          0       0       0       0         0           0           0   
407392          0       0       0       0         0           0           0   

        정신과개방상급병상수  정신과폐쇄개방병상수  격리병실병상수  무균치료실병상수  총의사수  의과일반의 인원수  의과인턴 인원수  \
0                0           0        0         0     1          0         0   
7                0           0        0         0     1          0         0   
17               0           0        0         0     2          0         0   
25               0           0        0         0     1          0         0   
36               0           0        0         0     1          0         0   
...            ...         ...      ...       ...   ...        ...       ...   
407363           0           0        0         0     1          0         0   
407371           0           0        0         0     1          0         0   
407376           0           0        0         0     1          0         0   
407384           0           0        0         0     1          0         0   
407392           0           0        0         0     1          0         0   

        의과레지던트 인원수  의과전문의 인원수  치과일반의 인원수  치과인턴 인원수  치과레지던트 인원수  치과전문의 인원수  \
0                0          1          0         0           0          0   
7                0          0          1         0           0          0   
17               0          2          0         0           0          0   
25               0          0 

In [16]:
df_2024_unique.to_csv('/Users/haeunpark/mid_project_data/2024_병원유형통합.csv', index=False)

In [17]:
# 암호화요양기호(병원 고유 식별자)를 기준으로 중복된 병원 제거 및 시도별 병원유형별 집계

# 중복제거
df_unique_hospitals = df_2024.drop_duplicates(subset=['암호화요양기호'])

# 집계
hospital_distribution_by_region = df_unique_hospitals.groupby(['시도코드명', '종별코드명']).size().unstack(fill_value=0)

# 서울, 경기, 인천을 가장 앞으로 두고 나머지 시도는 뒤에 위치하도록 설정
desired_order = ['서울', '경기', '인천'] + [x for x in hospital_distribution_by_region.index if x not in ['서울', '경기', '인천']]
hospital_distribution_by_region = hospital_distribution_by_region.reindex(desired_order)

# 인덱스를 다시 열로 변환 (reset_index)
hospital_distribution_by_region = hospital_distribution_by_region.reset_index()
hospital_distribution_by_region

종별코드명 시도코드명   병원  보건소  보건의료원  보건지소  보건진료소  상급종합  요양병원     의원  정신병원  조산원  종합병원  \
0        서울  225   25      0    10      0    14   109  10020    12    3    44   
1        경기  288   48      1   118    155     6   293   8301    53    5    66   
2        인천   59   10      0    27     25     3    63   1823    17    1    18   
3        강원   36   16      2    97    129     2    32    813     9    0    14   
4        경남  128   20      1   173    220     3   114   1727    30    1    23   
5        경북   52   22      2   216    298     0   104   1309    26    1    21   
6        광주   83    5      0     2     10     2    60   1068     9    0    23   
7        대구   90    9      0    15     18     5    73   2028    16    0    14   
8        대전   48    5      0     6      7     2    45   1135     7    1     8   
9        부산  145   16      0    11      5     4   161   2640    23    1    25   
10      세종시    3    1      0    10      7     0     4    231     1    0     2   
11       울산   32    5      0     8     11     1    37    645     6    0     8   
12       전남   73   19      3   216    327     1    79    963    11    0    25   
13       전북   57   11      4   148    240     2    75   1227    11    0    11   
14       제주    8    6      0    11     48     0    11    503     2    1     6   
15       충남   47   14      2   150    236     1    66   1136    17    0    12   
16       충북   37   13      1    94    159     1    36    933    12    1    12   

종별코드명  치과병원  치과의원  한방병원   한의원  
0        57  4897    89  3653  
1        44  4752   149  3352  
2        13  1004    48   688  
3         4   437     6   378  
4        22   954    27   785  
5        11   698     9   627  
6        14   645    88   319  
7        13   945    19   906  
8         7   558    15   501  
9        25  1338    27  1150  
10        1    99     3   101  
11        4   386     7   276  
12        7   514    29   355  
13        3   611    35   507  
14        1   247     1   196  
15       12   618    17   523  
16        4   475    12   409

In [18]:
hospital_distribution_by_region.to_csv('/Users/haeunpark/mid_project_data/2024_지역별병원유형수.csv', index=False)

In [19]:
# 그래프 생성
fig = px.bar(hospital_distribution_by_region, 
             x='시도코드명', 
             y=hospital_distribution_by_region.columns[1:],
             title='시도별 병원 유형 분포',
             barmode='stack')

fig.show()

# 시도별 병원 유형 분포 확인 결과

- 수도권: 의원, 치과의원과 같은 1차 의료기관의 수가 매우 많으며, 상급종합병원과 종합병원도 상대적으로 많은 수를 차지하고 있음.

- 비수도권: 지방은 서울, 경기, 인천에 비해 의원의 수는 상대적으로 적으나 요양병원이나 보건의료원, 보건지소와 같은 시설이 많이 분포되어 있음. 
이는 고령화 사회에서 중요하게 다뤄지는 요양 및 장기 치료 시설이 비수도권에 집중된 경향을 보여준다고 추측

- 세종시: 세종시는 상대적으로 의료기관의 숫자가 적지만, 이는 도시가 비교적 신생 도시이기며 공공기관이 몰려있는 곳임. 보건의료원 등의 공공 의료기관이 많이 활성화되어 있음.

# 시도별 1차, 2차, 3차 분포 차트

In [26]:
# 3차, 2차, 1차, 특수 의료로 분류하여 병원 유형 그룹화
hospital_distribution_by_region['3차 의료'] = hospital_distribution_by_region['상급종합']

hospital_distribution_by_region['2차 의료'] = (
    hospital_distribution_by_region['종합병원'] +
    hospital_distribution_by_region['병원'] +
    hospital_distribution_by_region['치과병원'] +
    hospital_distribution_by_region['한방병원'] +
    hospital_distribution_by_region['요양병원'] +
    hospital_distribution_by_region['정신병원'] +
    hospital_distribution_by_region['보건의료원']
)

# hospital_distribution_by_region['1차 의료'] = (
#     hospital_distribution_by_region['의원'] +
#     hospital_distribution_by_region['치과의원'] +
#     hospital_distribution_by_region['한의원'] +
#     hospital_distribution_by_region['보건소'] +
#     hospital_distribution_by_region['보건지소'] +
#     hospital_distribution_by_region['보건진료소']
# )

hospital_distribution_by_region['1차 의료'] = (
    hospital_distribution_by_region['의원'] +
    hospital_distribution_by_region['치과의원'] +
    hospital_distribution_by_region['한의원']
)

hospital_distribution_by_region['특수 의료'] = hospital_distribution_by_region['조산원']

hospital_grouped_ = hospital_distribution_by_region[['시도코드명', '3차 의료', '2차 의료', '1차 의료', '특수 의료']]

hospital_grouped_

종별코드명 시도코드명  3차 의료  2차 의료  1차 의료  특수 의료
0        서울     14    536  18570      3
1        경기      6    894  16405      5
2        인천      3    218   3515      1
3        강원      2    103   1628      0
4        경남      3    345   3466      1
5        경북      0    225   2634      1
6        광주      2    277   2032      0
7        대구      5    225   3879      0
8        대전      2    130   2194      1
9        부산      4    406   5128      1
10      세종시      0     14    431      0
11       울산      1     94   1307      0
12       전남      1    227   1832      0
13       전북      2    196   2345      0
14       제주      0     29    946      1
15       충남      1    173   2277      0
16       충북      1    114   1817      1

In [27]:
hospital_grouped_.to_csv('/Users/haeunpark/mid_project_data/2024_지역별_3차_2차_1차의료수_1차보건관련제외.csv', index=False)

In [28]:
fig = px.bar(hospital_grouped_, 
             x='시도코드명',  
             y=['3차 의료', '2차 의료', '1차 의료', '특수 의료'], 
             title='시도별 의료 서비스 수준 분포',
             barmode='stack')

# 그래프 표시
fig.show()

## 시도별 3차, 2차, 1차 의료 유형 분포 결과 분석

- 3차 의료: 서울과 경기는 3차 의료기관인 상급종합병원의 수가 상대적으로 많음. 
수도권의 중증 환자 치료와 관련한 전문 의료 서비스가 집중되어 있음을 나타냄.

- 2차 의료: 2차 의료기관(종합병원, 병원, 요양병원 등)의 경우 서울, 경기, 부산과 같은 대도시가 많으며, 이는 지방 거점 도시로서의 역할을 보여줍니다. 특히 경기는 894개의 2차 의료기관을 보유하고 있어 수도권의 중심 도시로서 중증 및 경증 환자들을 폭넓게 수용하고 있음.

- 1차 의료: 1차 의료기관(의원, 보건소, 치과의원 등)은 서울, 경기가 여전히 높은 수를 보이며, 이외 대부분의 시도에는 고르게 분포되어 있음.

- 특수의료(조산원): 특수 의료기관인 조산원은 소수의 시도에만 존재하며, 주로 출산 관련 인프라가 큰 이슈가 되는 지역에 집중되어 있음. 서울, 경기, 인천을 제외하고는 거의 없는 상태.

## 인사이트

### 1. 의료 서비스의 수도권 집중:
3차 의료 서비스는 서울과 경기에 집중되어 있음. 이로 인해 중증 환자들이 수도권에 몰리는 현상이 발생할 가능성이 큽니다. 이는 수도권의 병원이 과밀화될 위험을 초래되며, 지방 의료 서비스의 불균형이 이루어짐.

### 2. 1차 의료 서비스의 전국적 분포:
1차 의료기관은 전국적으로 고르게 분포되어 있음 특히 지방에서는 의원 및 보건소를 통해 지역 주민들이 기본적인 의료 서비스를 이용할 수 있는 환경이 조성되어 있음. 이는 지역 주민의 기본적인 의료 접근성이 어느 정도 보장되고 있음을 말함.

### 3. 지방의 2차 의료기관 역할 강화 필요:
지방의 경우 2차 의료기관의 수가 수도권보다 적지만, 지방 거점 도시들(부산, 대구 등)에 나름의 분포를 보이고 있음. 
그러나 이는 아직 수도권에 비해 부족한 수준임. 지방 중소도시에서도 중증 환자를 수용할 수 있는 2차 의료기관의 확충이 필요함.

### 4. 고령화에 따른 요양병원과 장기 요양 서비스:
요양병원은 지방에 많이 분포되어 있음. 이는 고령화 사회에서 중요한 역할을 함.
특히 농촌 지역에서 장기 요양 서비스의 수요가 높다는 점?..일까?  이는 지방에서 고령화 대책이 중점적으로 다뤄져야 함을 보여줌.

###  5. 특수의료 서비스의 부족:
조산원과 같은 특수 의료 서비스는 매우 제한적으로 존재하고 있으며, 이는 출산 인프라의 격차를 보여줌. 
특히 지방에서 출산 관련 서비스의 접근성이 떨어질 수 있음을 말하며,  산모 건강과 출산율에 영향을 미칠 수 있는 중요한 요인임.